processors/blockprocessor/block_processor.go

```diff
@@ -27,8 +29,112 @@ type blockProcessor struct {
 	logger  *log.Logger
 }
 
-// MakeProcessorWithLedger creates a block processor with a given ledger
-func MakeProcessorWithLedger(logger *log.Logger, l *ledger.Ledger, handler func(block *ledgercore.ValidatedBlock) error) (processor.Processor, error) {
+// Constructor is the ProcessorConstructor implementation for the "block processor" processor
+type Constructor struct{}
+
+// New initializes a postgresqlExporter
+func (c *Constructor) New() processors.Processor {
+	return &blockProcessor{}
+}
+
+var implName string = "Block Processor"
+
+func (proc *blockProcessor) Metadata() processors.ProcessorMetadata {
+	return processors.ProcessorMetadata{
+		ImpName:        implName,
+		ImpDescription: "Local Ledger Block Processor",
+		ImpDeprecated:  false,
+	}
+}
+
+func (proc *blockProcessor) Config() plugins.PluginConfig {
+	return ""
+}
+
+func (proc *blockProcessor) Init(ctx context.Context, cfg plugins.PluginConfig) error {
+	processors.RegisterProcessor(implName, &Constructor{})
+	return nil
+}
+
+func (proc *blockProcessor) Close() error {
+	return nil
+}
+
+func (proc *blockProcessor) Process(input data.BlockData) (data.BlockData, error) {
```

Comment:
This looks a lot like `ProcessBlockCert`, why aren't you using the interface?

Reply:
I've consolidated a lot of common functionality.  I've maintained `ProcessBlockCert` to give us backwards functionality with the current Indexer as a stepping stone.